### ISSUE TITLE
C.12 Question

### ISSUE THREAD

**Issue by CompCodeNet** [isocpp/CppCoreGuidelines#1950]
If use pointer, is C.12 saying to use non-const pointer? 

I'm referring to const of the pointer itself, not the object that is pointed to:
I.e. `ClassXyz* const _ptr` not `const ClassXyz* _ptr`

Are we to understand that the idea is to lose the accidental-member-overwrite protection that a const provides for the sake of the copyability benefits of C.12? 

As far as usefulness of const data members, I respectfully disagree. I have a use case. Also, even the pointer use case is considered valid by C.12 and if the pointer member is useful then ability to freeze pointer value (by using const) is useful and less error prone.

**Comment by beinhaerter** [comment]
Your understanding of the rule is right.

Your question is not new, there have been [multiple discussions about C.12](https://github.com/isocpp/CppCoreGuidelines/issues?q=is%3Aissue+%22c.12%22).

**Comment by CompCodeNet** [comment]
@beinhaerter thank you. Before posting, I did look at all C.12 specific posts, but I do see now that the other post(s) touch on what I am referring to. 

Therefore, one could decide to use a non-const pointer data member (for example) for the sake of copyability? This would be with the understanding that it is a class member after all, and so even if not const, it is limited to the class and therefore it does have that level of protection against overwrites - i.e., as for all your class members, don't inadvertently step on it.

Perhaps the language needs a "const except for special member functions" keyword or similar idea.

(I came back to C++ full-time about a year ago, having used C/C++ in the late 80s and 90s. I'm impressed with C++20 and the core guidelines and I follow them. span and string_view are excellent and it's also great to see the format library. Looking forward to C++20 modules maturity.)

Thanks much for response.
